### ISSUE TITLE
Validate files that are missing from code coverage

### ISSUE THREAD

**Issue by chimon2000** [VeryGoodOpenSource/very_good_coverage#34]
**Is your feature request related to a problem? Please describe.**

Currently, running flutter test --coverage generates a report that does not include files w/o test coverage.  This causes the code coverage reporting to be skewed. 

**Describe the solution you'd like**

Test coverage verification should also validate files missing from coverage.

**Describe alternatives you've considered**

Currently, I run a script that will generate an empty test that imports every dart file (with some exclusions).  

```bash
#!/bin/sh

outputFile="$(pwd)/test/coverage_test.dart"
packageName="$(cat pubspec.yaml| grep '^name: ' | awk '{print $2}')"

if [ "$packageName" = "" ]; then
    echo "Run $0 from the root of your Dart/Flutter project"
    exit 1
fi

echo "/// *** GENERATED FILE - ANY CHANGES WOULD BE OBSOLETE ON NEXT GENERATION *** ///\n" > "$outputFile"
echo "/// Helper to test coverage for all project files" >> "$outputFile"
echo "// ignore_for_file: unused_import" >> "$outputFile"
find lib -name '*.dart' | grep -v '.g.dart' | grep -v 'generated_plugin_registrant' | awk -v package=$packageName '{gsub("^lib", "", $1); printf("import '\''package:%s%s'\'';\n", package, $1);}' >> "$outputFile"
echo "\nvoid main() {}" >> "$outputFile"
```

This is less than ideal because it requires Dart developers to have a familiarity with bash.  Also, adding to the exclude list could become tedious.

**Additional context**

n/a

**Comment by jorgecoca** [comment]
I'm torn about this solution; I agree that this is a problem, but I don't think this should handled by this action, as it should be platform agnostic. After all, `very-good-coverage` only works with the `lcov` file you give it.

I think this bug should be raised to the coverage tool of flutter: IMO, that tool should be the one that registers all the files and passes them to the lcov report. 

Thoughts?

**Comment by felangel** [comment]
> I'm torn about this solution; I agree that this is a problem, but I don't think this should handled by this action, as it should be platform agnostic. After all, `very-good-coverage` only works with the `lcov` file you give it.
> 
> I think this bug should be raised to the coverage tool of flutter: IMO, that tool should be the one that registers all the files and passes them to the lcov report.
> 
> Thoughts?

I agree, `very-good-coverage` should not be flutter/dart specific and it should just rely on an `lcov` file as input. We can probably raise an issue on the flutter tools to do something similar to what [test_coverage](https://pub.dev/packages/test_coverage) does and generate a main which imports the entrypoint (`main.dart`).

I think we should avoid importing every file because an added benefit of code coverage is you can find and remove unused code 😄

**Comment by alestiago** [comment]
I agree with @felangel, Very Good Coverage is meant to be agnostic from the actual development environment and only cares about a generated lcov file. Your use-case is specific to Flutter, perhaps this might be a good issue for Very Good Workflows.

Please, let me know if this helps.

**Comment by alestiago** [comment]
Closing this issue since it was opened a few years ago and there doesn't seem to be involvement from @chimon2000 since.  

Although this issue is clearly not an issue in Very Good Coverage it is something we would like to provide a defined solution to as part of a Very Good Open Source product. Since it has caused developers to think their coverage is 100% when using the dart of flutter command line interfaces to generate the `lcov.info` file.

We will be closing this in favour of https://github.com/VeryGoodOpenSource/very_good_cli/issues/464.
